src/core/lib/transport/handshaker_factory.h

```diff
@@ -35,11 +35,37 @@ namespace grpc_core {
 
 class HandshakeManager;
 
+// Enum representing the priority of the handshakers.
+// The order of the handshakers is decided by the priority.
+// For example kPreTCPConnect handshakers are called before kTCPConnect and so
+// on.
+enum class HandshakerPriority : int {
```

Comment:
Suggest moving this inside of the `HandshakerFactory` class.

Reply:
Done.